model/Test/Normalizer/TestNormalizer.php

```diff
@@ -46,17 +45,17 @@ class TestNormalizer
     ];
 
     private QtiTestService $qtiTestService;
-    private IndexService $indexService;
+    private IndexDocumentBuilderInterface $legacyDocumentBuilder;
 
-    public function __construct(QtiTestService $qtiTestService, IndexService $indexService)
+    public function __construct(QtiTestService $qtiTestService, IndexDocumentBuilderInterface $legacyDocumentBuilder)
```

Comment:
Same as here https://github.com/oat-sa/extension-tao-advanced-search/pull/108/files#r1335851449 We can leverage to use `AdvancedSearchIndexDocumentBuilder` only now.

Reply:
if we call `AdvancedSearchIndexDocumentBuilder` from https://github.com/oat-sa/extension-tao-advanced-search/blob/develop/model/Test/Normalizer/TestNormalizer.php#L57 we will end up with infinite loop here https://github.com/oat-sa/extension-tao-advanced-search/blob/develop/model/Index/Service/AdvancedSearchIndexDocumentBuilder.php#L80

Reply:
Ouch. True! I hope the other places I mentioned we do not have this issue 🤞